code-studio/src/js/utils.js

```diff
@@ -0,0 +1,28 @@
+var utils = module.exports;
+
+/**
+ * Wrap a function so that it will only be executed a certain amount of time
+ * after it's originally called, to avoid multiple calls in quick succession.
+ * Subsequent calls to the returned function will reset its timer.
+ * Note: Would reuse lodash's debounce method here, but we don't appear to
+ * have access to it here.
+ * @param {function} callback - to be called after the given delay
+ * @param {number} delay - How long to wait after the _last_ call before executing the callback
+ * @returns {function}
+ */
+utils.debounce = function (callback, delay) {
```

Comment:
The _only_ method in here is `debounce`... can we replace this with lodash debounce now?